File: app/src/main/java/wjy/yo/ereader/entity/userdata/Preference.java
package wjy.yo.ereader.entity.userdata;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;

import lombok.Data;
import lombok.EqualsAndHashCode;
import wjy.yo.ereader.entity.UserData;

@Entity(tableName = "user_preference", indices = {@Index(value = {"userName", "code"}, unique = true)})
@Data
@EqualsAndHashCode(callSuper = true)
public class Preference extends UserData {

    private String code;

    private String value;
}

File: app/src/main/java/wjy/yo/ereader/entity/userdata/Preference.kt
package wjy.yo.ereader.entity.userdata

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Index

import wjy.yo.ereader.entity.UserData

@Entity(tableName = "user_preference", indices = arrayOf(Index(value = arrayOf("userName", "code"), unique = true)))
class Preference(var code: String) : UserData() {

    var value: String? = null
}
